doc/basics.md

```diff
@@ -541,6 +544,56 @@ index allows you to select the indexed node. Within objects, the string value of
 select the value. If your keys contain the characters '/' or '~', they must be escaped as '~1' and
 '~0' respectively. An empty JSON Path refers to the whole document.
 
+For multiple JSON pointer queries on a document, one can call `at_pointer` multiple times.
+
+```c++
+auto cars_json = R"( [
+  { "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
+  { "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
+  { "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
+] )"_padded;
+ondemand::parser parser;
+auto cars = parser.iterate(cars_json);
+size_t size = cars.count_elements();
+
+for (size_t i = 0; i < size; i++) {
+    std::string json_pointer = "/" + std::to_string(i) + "/tire_pressure/1";
+    double x = cars.at_pointer(json_pointer);
+    std::cout << x << std::endl; // Prints 39.9, 31 and 30
+}
+```
+
+It is important to note that `at_pointer` calls `rewind` only  at the start of every call, before parsing along the JSON path (this is what allows to make multiple JSON pointer queries without manually resetting the parser to point at the beginning of the document). See more about rewind [here](https://github.com/simdjson/simdjson/blob/master/doc/basics.md#rewinding). For example, the following would not work:
+
+```c++
+auto json = R"( {
+  "k0": 27,
+  "k1": [13,26],
+  "k2": true
+} )"_padded;
+ondemand::parser parser;
+auto doc = parser.iterate(json);
+
+std::cout << doc.at_pointer("/k1/1") << std::endl; // Prints 26
+std::cout << doc.at_pointer("/k2") << std::endl; // Prints true
+std::cout << doc.find_field("k0") << std::endl; // Does not work because doc is not rewinded
+```
+But the following would work:
+```c++
+auto json = R"( {
+  "k0": 27,
+  "k1": [13,26],
+  "k2": true
+} )"_padded;
+ondemand::parser parser;
+auto doc = parser.iterate(json);
+
+std::cout << doc.at_pointer("/k1/1") << std::endl; // Prints 26
+std::cout << doc.at_pointer("/k2") << std::endl; // Prints true
+doc.rewind();
+std::cout << doc.find_field("k0") << std::endl; // Prints 27
+```
```

Comment:
I suggest also adding the example from https://github.com/simdjson/simdjson/pull/1625 or something simpler, to illustrate how one has to consume the content from the at_pointer call (one should not just keep the references to values).